NSS-Day-Cohort-53/tabloid-full-stack-always-puffy-impalas
# Users may subscribe to other users

#### Changes Made
1. Added file `subscriptionManager` to `/tabloid/client/src/modules` directory.
2. Added file `SubscriptionRepository` & `ISubscriptionRepository` to `/tabloid/repositories` directory.
3. Added file `Subscription` to `/tabloid/models` directory.
4. Added file `SubscriptionController` to `/tabloid/controllers` directory.
5. Modified file `postDetails` to include `doSubscribe()` that handles subscription clicks.
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout cg-userSubscriptions
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.
    > Log in if not already logged in
    > Navigate to post list and click to view a post's details
    > If your database is fresh, you do not have any subscriptions. Any post's details viewed will show the option to subscribe to an author
    > Click to subscribe to the author. A database entry will be created for you and the DOM will change 'subscribe' to 'unsubscribe'
    > If user clicks unsubscribe, nothing will happen because that is a separate issue
4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

Closes #18
